astropy/coordinates/baseframe.py

```diff
@@ -789,6 +790,11 @@ def _get_representation_info(cls):
         # This exists as a class method only to support handling frame inputs
```

Comment:
Seeing this, I wondered generally whether this still needed to be a class method. A quick test showed no failures at all when one just moves this to the `representation_info` property. Hence, it feels like this should be separated correctly in parts that are class and self-specific.